src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -0,0 +1,121 @@
+                                summary='Unable to export helm chart from the registry')
+        raise CLIError("Unable to export helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_export.decode("ascii"))
+
+
+def add_helm_repo(kube_config, kube_context):
+    if os.getenv('HELMREPONAME') and os.getenv('HELMREPOURL'):
+        repo_name = os.getenv('HELMREPONAME')
+        repo_url = os.getenv('HELMREPOURL')
+        cmd_helm_repo = ["helm", "repo", "add", repo_name, repo_url, "--kubeconfig", kube_config]
+        if kube_context:
+            cmd_helm_repo.extend(["--kube-context", kube_context])
+        response_helm_repo = Popen(cmd_helm_repo, stdout=PIPE, stderr=PIPE)
+        _, error_helm_repo = response_helm_repo.communicate()
+        if response_helm_repo.returncode != 0:
+            telemetry.set_exception(exception=error_helm_repo.decode("ascii"), fault_type=consts.Add_HelmRepo_Fault_Type,
+                                    summary='Failed to add helm repository')
+            raise CLIError("Unable to add repository {} to helm: ".format(repo_url) + error_helm_repo.decode("ascii"))
+
+
+def get_helm_registry(profile, location):
+    cred, _, _ = profile.get_login_credentials(
+        resource='https://management.core.windows.net/')
+    token = cred._token_retriever()[2].get('accessToken')  # pylint: disable=protected-access
+
+    get_chart_location_url = "https://{}.dp.kubernetesconfiguration.azure.com/{}/GetLatestHelmPackagePath?api-version=2019-11-01-preview".format(location, 'azure-arc-k8sagents')
+    query_parameters = {}
+    query_parameters['releaseTrain'] = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
+    header_parameters = {}
+    header_parameters['Authorization'] = "Bearer {}".format(str(token))
+    try:
+        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                                summary='Error while fetching helm chart registry path')
+        raise CLIError("Error while fetching helm chart registry path: " + str(e))
+    if response.status_code == 200:
+        return response.json().get('repositoryPath')
+    telemetry.set_exception(exception=str(response.json()), fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
+                            summary='Error while fetching helm chart registry path')
+    raise CLIError("Error while fetching helm chart registry path: {}".format(str(response.json())))
+
+
+def is_update_allowed(agent_version):
+    release_train = os.getenv('RELEASETRAIN') if os.getenv('RELEASETRAIN') else 'stable'
+    if release_train == "stable" and agent_version >= "0.2.5":
+        return True
+
+    if release_train == "dev" and agent_version >= "0.1.214-dev":
+        return True
+
+    return False
```

Comment:
Why do we need this? Is this because proxy values were added in 0.2.5 or 0.1.214-dev and we want to allow update only after that?

Even if the user was running an agent version lower than 0.2.5 or 0.1.214 and ran the update command later, it should still work fine right? Because we'd just be running helm upgrade with reuse values and additional values set for the proxy flags.

Also why 0.2.5? I see HISTORY.rst and setup.py got updated to 0.2.3